x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_page.tsx

```diff
@@ -3,38 +3,41 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import React, { memo, useCallback, useMemo } from 'react';
-import { FormattedMessage } from '@kbn/i18n/react';
-import { EuiButton, EuiButtonEmpty } from '@elastic/eui';
-import { useHistory, useLocation } from 'react-router-dom';
+import React, { memo, useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
 import styled from 'styled-components';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { ViewType } from '../state';
+import { getCurrentLocation, getListTotalItemsCount } from '../store/selectors';
+import { useTrustedAppsNavigateCallback, useTrustedAppsSelector } from './hooks';
 import { AdministrationListPage } from '../../../components/administration_list_page';
+import { CreateTrustedAppFlyout } from './components/create_trusted_app_flyout';
+import { ControlPanel } from './components/control_panel';
+import { TrustedAppsGrid } from './components/trsuted_apps_grid';
 import { TrustedAppsList } from './trusted_apps_list';
 import { TrustedAppDeletionDialog } from './trusted_app_deletion_dialog';
 import { TrustedAppsNotifications } from './trusted_apps_notifications';
-import { CreateTrustedAppFlyout } from './components/create_trusted_app_flyout';
-import { getTrustedAppsListPath } from '../../../common/routing';
-import { useTrustedAppsSelector } from './hooks';
-import { getCurrentLocation } from '../store/selectors';
 import { TrustedAppsListPageRouteState } from '../../../../../common/endpoint/types';
 import { useNavigateToAppEventHandler } from '../../../../common/hooks/endpoint/use_navigate_to_app_event_handler';
 
 export const TrustedAppsPage = memo(() => {
-  const history = useHistory();
   const { state: routeState } = useLocation<TrustedAppsListPageRouteState | undefined>();
   const location = useTrustedAppsSelector(getCurrentLocation);
-  const handleAddButtonClick = useCallback(() => {
-    history.push(
-      getTrustedAppsListPath({
-        ...location,
-        show: 'create',
-      })
-    );
-  }, [history, location]);
-  const handleAddFlyoutClose = useCallback(() => {
-    const { show, ...paginationParamsOnly } = location;
-    history.push(getTrustedAppsListPath(paginationParamsOnly));
-  }, [history, location]);
+  const totalItemsCount = useTrustedAppsSelector(getListTotalItemsCount);
+  const handleAddButtonClick = useTrustedAppsNavigateCallback(() => ({ show: 'create' }));
```

Comment:
I'm having a hard time understanding this. how is `useTrustedAppsNavigateCallback()` memoizing? its getting a new function on input here every render cycle.